src/ir.cpp

```diff
@@ -8488,6 +8488,20 @@ static ConstCastOnly types_match_const_cast_only(IrAnalyze *ira, TypeTableEntry
         return result;
     }
 
+    // *T and [*]T can always cast to ?*c_void and ?[*]c_void
+    if (wanted_type->id == TypeTableEntryIdPointer &&
+        wanted_type->data.pointer.child_type == g->builtin_types.entry_c_void &&
+        actual_type->id == TypeTableEntryIdPointer &&
+        (!actual_type->data.pointer.is_const || wanted_type->data.pointer.is_const) &&
+        (!actual_type->data.pointer.is_volatile || wanted_type->data.pointer.is_volatile) &&
+        // Do we want to allow *T -> [*]c_void?
```

Comment:
I don't think `[*]c_void` is possible, since zig does not allow unknown length pointer to opaque type:

```zig
test "aoeu" {
    _ = [*]c_void;
}
```
```
/home/andy/dev/zig/build/test.zig:2:9: error: unknown-length pointer to opaque
    _ = [*]c_void;
        ^
```